src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -462,69 +394,33 @@ private SinkRecord handleNativeRecord(SinkRecord record, boolean isKey) {
 
   @Override
   public void insertBufferedRecordsIfFlushTimeThresholdReached() {
-    if (this.streamingBufferThreshold.shouldFlushOnBufferTime(this.previousFlushTimeStampMs)) {
-      LOGGER.debug(
-              "Time based flush for channel:{}, CurrentTimeMs:{}, previousFlushTimeMs:{},"
-                      + " bufferThresholdSeconds:{}",
-              this.getChannelNameFormatV1(),
-              System.currentTimeMillis(),
-              this.previousFlushTimeStampMs,
-              this.streamingBufferThreshold.getFlushTimeThresholdSeconds());
-      StreamingBuffer copiedStreamingBuffer;
-      bufferLock.lock();
-      try {
-        copiedStreamingBuffer = this.streamingBuffer;
-        this.streamingBuffer = new StreamingBuffer();
-      } finally {
-        bufferLock.unlock();
-      }
-      if (copiedStreamingBuffer != null) {
-        insertRecords(copiedStreamingBuffer);
-      }
-    }
+    // It is just a leftover after buffered channel, not needed here as a buffer was removed in the current class.
+    // todo remove this method in the future
   }
 
-  public InsertRowsResponse insertRecords(StreamingBuffer streamingBufferToInsert) {
-    // intermediate buffer can be empty here if time interval reached but kafka produced no records.
-    if (streamingBufferToInsert.isEmpty()) {
-      LOGGER.debug("No Rows Buffered for channel:{}, returning", this.getChannelNameFormatV1());
-      this.previousFlushTimeStampMs = System.currentTimeMillis();
-      return null;
-    }
-    InsertRowsResponse response = null;
+  public void transformAndSend(SinkRecord kafkaSinkRecord) {
```

Comment:
Why is it public?

Reply:
If it's complex enough, maybe it deserves to be a separate class. Definitely, we should not make methods public for the sake of testing them. The class should protect its private state thus we can only test them using their public interface.

I changed it to private now.